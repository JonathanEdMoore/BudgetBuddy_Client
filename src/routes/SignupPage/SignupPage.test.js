import React from 'react'
import ReactDOM from 'react-dom'
import SignupPage from './SignupPage'
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('SignupPage smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><SignupPage /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('SignupPage snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<SignupPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})