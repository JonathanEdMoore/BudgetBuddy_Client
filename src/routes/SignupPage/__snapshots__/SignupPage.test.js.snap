// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage snapshot test renders the UI as expected 1`] = `
<div>
  <main
    role="main"
  >
    <header>
      <MainHeader />
      <h2>
        Sign Up
      </h2>
    </header>
    <section>
      <BudgetBuddyForm
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <label
            htmlFor="first-name"
          >
            First name
            <p
              className="error"
            />
          </label>
          <input
            id="first-name"
            name="first-name"
            onChange={[Function]}
            placeholder="First Name"
            type="text"
          />
          <ValidationError
            hasError={true}
            message=""
          />
        </div>
        <div
          className="field"
        >
          <label
            htmlFor="last-name"
          >
            Last name
            <p
              className="error"
            />
          </label>
          <input
            id="last-name"
            name="last-name"
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
          />
          <ValidationError
            hasError={true}
            message=""
          />
        </div>
        <div
          className="field"
        >
          <label
            htmlFor="username"
          >
            Email
            <p
              className="error"
            />
          </label>
          <input
            id="email"
            name="email"
            onChange={[Function]}
            type="text"
          />
          <ValidationError
            hasError={true}
            message=""
          />
        </div>
        <div
          className="field"
        >
          <label
            htmlFor="password"
          >
            Password
            <p
              className="error"
            />
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
          />
          <p
            className="submission-error"
          />
          <ValidationError
            hasError={true}
            message=""
          />
        </div>
        <button
          disabled={true}
          type="submit"
        >
          Sign Up
        </button>
      </BudgetBuddyForm>
      <p>
        Already have an account?
        <Link
          to="/login"
        >
          <button>
            Log in
          </button>
        </Link>
      </p>
    </section>
  </main>
</div>
`;
